src/server/scripts/Outland/HellfireCitadel/MagtheridonsLair/boss_magtheridon.cpp

```diff
@@ -16,617 +15,590 @@
-            uint32 trigger;
-
-            void Reset() override
+            void EnterEvadeMode(EvadeReason /*why*/) override
             {
-                Initialize();
+                summons.DespawnAll();
+                events.Reset();
+                instance->SetData(DATA_MANTICRON_CUBE, ACTION_DISABLE);
+                instance->SetData(DATA_COLLAPSE, ACTION_DISABLE);
+                instance->SetData(DATA_COLLAPSE_2, ACTION_DISABLE);
+
+                _DespawnAtEvade();
             }
 
-            void SpellHit(Unit*, const SpellInfo* spell) override
+            void SummonedCreatureDies(Creature* summon, Unit* /*killer*/) override
             {
-                if (trigger == 2 && spell->Id == SPELL_BLAZE_TARGET)
+                if (summon->GetEntry() == NPC_HELLFIRE_CHANNELLER && events.IsInPhase(PHASE_1))
                 {
-                    DoCast(me, SPELL_BLAZE_TRAP, true);
-                    me->SetVisible(false);
-                    Despawn_Timer = 130000;
+                    _channelersCount--;
+
+                    if (_channelersCount == 0)
+                        CombatStart();
                 }
             }
 
-            void SetTrigger(uint32 _trigger)
+            void DoAction(int32 action) override
             {
-                trigger = _trigger;
-                me->SetDisplayId(11686);
-                if (trigger == 1) //debris
+                if (action == ACTION_START_CHANNELERS_EVENT && events.IsInPhase(PHASE_BANISH))
                 {
-                    me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
-                    DoCast(me, SPELL_DEBRIS_VISUAL, true);
-                    FireBlast_Timer = 5000;
-                    Despawn_Timer = 10000;
+                    events.SetPhase(PHASE_1);
+                    Talk(EMOTE_WEAKEN, me);
+                    summons.DoZoneInCombat(NPC_HELLFIRE_CHANNELLER);
+                    events.ScheduleEvent(EVENT_START_FIGHT, Minutes(2));
+                    events.ScheduleEvent(EVENT_NEARLY_EMOTE, Seconds(60));
+                    instance->SetBossState(DATA_MAGTHERIDON, IN_PROGRESS);
+                    instance->SetData(DATA_CALL_WARDERS, ACTION_ENABLE);
                 }
             }
 
-            void EnterCombat(Unit* /*who*/) override
+            void SpellHit(Unit* /*caster*/, SpellInfo const* spell) override
             {
-                DoZoneInCombat();
+                if (spell->Id == SPELL_SHADOW_CAGE)
+                    Talk(SAY_BANISHED);
             }
-            void AttackStart(Unit* who) override
+
+            void DamageTaken(Unit* /*who*/, uint32& damage) override
             {
-                if (!trigger)
-                    ScriptedAI::AttackStart(who);
+                if (me->HealthBelowPctDamaged(30, damage) && !events.IsInPhase(PHASE_3))
+                {
+                    events.SetPhase(PHASE_3);
+                    me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
```

Comment:
not exactly.
I saw in videos, when he change for phase 3, players/pets cant attack him
On my sniffs, Hunter and Pet stop attack before he cast camera spell..
for me, the easy way to reproduce it is with IMMUNE_TO_PC flag.
and in sniffs and in videos, dont show any "immune damage" in this phase, only players cant attack him